### PR TITLE
[TwigBridge] Allow floats in html5 input type number field

### PR DESCRIPTION
**Pull request by wimhendrikx**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | yes
| New feature?  |no
| Deprecations? | no
| Tickets       | 
| License       | MIT
| Doc PR        | 

Adding the attribute step: any allows for correct use of floats in number input

### CODE REVIEW

**Review by nicolas-grekas — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Can you please have a look a failing tests?

**Review by xabbuh — CHANGES_REQUESTED** [review]
This does not work with localized numbers, does it? At least from what I see the arrows rendered by the browser will change the format of the rendered values.

**Comment by wimhendrikx** [comment]
Hi @xabbuh ,

It changes the format to the used locale notation. For me it keeps `,` and changes `.` to `,` like it should.

**Comment by wimhendrikx** [comment]
Hi @nicolas-grekas,

I fixed the issues in the tests and now it works for 7.2, 7.4 and 8.2 but not for 8.0 and 8.2. Any idea how I can fix it for the other 2?

File: src/Symfony/Bridge/Twig/Resources/views/Form/form_div_layout.html.twig [hunk 1]
```diff
@@ -172,8 +172,8 @@
 {%- endblock dateinterval_widget -%}
 
 {%- block number_widget -%}
-    {# type="number" doesn't work with floats in localized formats #}
-    {%- set type = type|default('text') -%}
+    {%- set type = type|default('number') -%}
+    {%- set attr = attr|merge({step: 'any'}) -%}
```
Comment (yceruto) [inline src/Symfony/Bridge/Twig/Resources/views/Form/form_div_layout.html.twig]: The `step` attribute should only be used when the `type` is set to `number`, right?

**Review by yceruto — COMMENTED** [review]
It works well on my side too.

**Comment by esserj** [comment]
isn't this contradicting the html5 option of the NumberType

```php
#\Symfony\Component\Form\Extension\Core\Type\NumberType
    public function buildView(FormView $view, FormInterface $form, array $options)
    {
        if ($options['html5']) {
            $view->vars['type'] = 'number';
        }
    }
```
The default is assuming the same default as the Type is, html5 is false by default, and thus is opted for text instead?

**Comment by wimhendrikx** [comment]
@esserj , thanks for pointing that out. That option is not needed anymore so I will remove it.

**Comment by wimhendrikx** [comment]
I'm gonna change this PR and put the steps: any attribute in the html5 option. This way it is fully BC.

File: src/Symfony/Bridge/Twig/Resources/views/Form/form_div_layout.html.twig (hunk 1 above)
Comment (wimhendrikx) [reply src/Symfony/Bridge/Twig/Resources/views/Form/form_div_layout.html.twig]: Yes good point, I'm gonna move it to the html5 option as talked beneath.

**Comment by nicolas-grekas** [comment]
Thank you @wimhendrikx.

**Pull request merged**